src/uct/base/uct_mem.c

```diff
@@ -111,10 +138,12 @@ ucs_status_t uct_mem_alloc(size_t min_length, unsigned flags, uct_alloc_method_t
             break;
 
         case UCT_ALLOC_METHOD_MMAP:
+            map_flags = mmap_flags(flags);
             /* Request memory from operating system using mmap() */
             alloc_length = ucs_align_up_pow2(min_length, ucs_get_page_size());
-            address = ucs_mmap(NULL, alloc_length, PROT_READ|PROT_WRITE,
-                               MAP_PRIVATE|MAP_ANON, -1, 0 UCS_MEMTRACK_VAL);
+            address = ucs_mmap((map_flags & MAP_FIXED) ? mem->address : NULL,
```

Comment:
@yosefe  This will affect a lot of test codes (and probably applications) because it means that address must be always initialized and this field is only one which has such requirement.
Before this PR *mem was strongly out parameter in uct_mem_alloc function.
May be to add another flag like UCT_MD_MEM_FLAG_HINT for backward compatibility?
The same is actual for UCP. For now, the implementation is valid only because hint do not guarantee the result :)